internal/configuration/configuration.go

```diff
@@ -80,33 +83,39 @@ func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
 	configurationEqual := reflect.DeepEqual(message.Configuration, m.deviceConfiguration.Configuration)
 	workloadsEqual := reflect.DeepEqual(message.Workloads, m.deviceConfiguration.Workloads)
 	log.Tracef("Initial config: [%v]; workloads equal: [%v]; configurationEqual: [%v]", m.IsInitialConfig(), workloadsEqual, configurationEqual)
+	var errors error
+
 	if m.IsInitialConfig() || !(configurationEqual && workloadsEqual) {
 		log.Tracef("Updating configuration: %v", message)
 		for _, observer := range m.observers {
 			err := observer.Update(message)
 			if err != nil {
-				return err
+				errors = multierror.Append(errors, fmt.Errorf("cannot update observer: %s", err))
+				return errors
 			}
 		}
 
 		// TODO: handle all the failure scenarios correctly; i.e. compensate all the changes that has already been introduces.
 		file, err := json.MarshalIndent(message, "", " ")
 		if err != nil {
-			return err
+			errors = multierror.Append(errors, fmt.Errorf("cannot unmarshal JSON: %s", err))
+			return errors
 		}
 		log.Tracef("Writing config to %s: %v", m.deviceConfigFile, file)
 		err = ioutil.WriteFile(m.deviceConfigFile, file, 0640)
 		if err != nil {
 			log.Error(err)
-			return err
+			errors = multierror.Append(errors, fmt.Errorf("cannot write device config file '%s': %s", m.deviceConfigFile, err))
+			return errors
```

Comment:
Same here; probably no return at this stage.

Reply:
Discussed offline, I'll address this on a new PR regarding configuration tests